pingpong/rdm_cntr_pingpong.c

```diff
@@ -91,18 +91,18 @@ static int recv_xfer(int size)
 {
 	int ret;
 
-	ret = fi_cntr_wait(rxcntr, recv_outs, -1);
-	if (ret < 0) {
-		FT_PRINTERR("fi_cntr_wait", ret);
-		return ret;
-	}
-
 	ret = fi_recv(ep, rx_buf, rx_size, fi_mr_desc(mr), remote_fi_addr,
 			&rx_ctx);
 	if (ret)
 		FT_PRINTERR("fi_recv", ret);
 	recv_outs++;
 
+	ret = fi_cntr_wait(rxcntr, recv_outs, -1);
+	if (ret < 0) {
+		FT_PRINTERR("fi_cntr_wait", ret);
+		return ret;
+	}
+
```

Comment:
fi_cntr_wait() is being used for completions.  Due to the ordering (cntr_wait() then fi_recv()).  the old code effectively doesn't wait for a completion for the final recv.  When the client exits before completing it's final receive, the send side can't complete a rendezvous send.  The test can't hang when using eager path message sizes in GNI.  I assume other providers don't rely on receive side progress like we do with rendezvous.